### DISCUSSION TITLE
Unsafe.As, readonly version

### DISCUSSION THREAD

**Discussion by zugolbr** [dotnet/runtime#54789] [Ideas] [answered]
Hi, we have
`ref TTo Unsafe.As<TFrom, TTo>(ref TFrom)`

Is there anything preventing
`ref readonly TTo Unsafe.ReadOnlyAs<TFrom, TTo>(ref readonly TFrom)`

It would be quite handy... :)

**Comment by huoyaoyuan** [comment, marked as answer]
> It would be quite handy... :)

I don't think so.

You can do much less thing with readonly ref than regular ref. All interlocked or volatile apis accept `ref`.

Currently there is a workaround `Unsafe.As(Unsafe.AsRef())`. Since this is unsafe, I don't think we should make a shortcut.

**Reply by zugolbr** [reply]
Well, thanks for pointing it out! I'm ok with the workaround... :)
